src/Persistance/RepositoryImplementations/SystemRegisterRepository.cs

```diff
@@ -414,4 +415,21 @@ private static List<Right> GetRights(string[] rights)
     {
         return JsonSerializer.Deserialize<List<Right>>(rights[0]);
     }
+
+    private static string? GetOrgNumber(IDictionary<string, string> vendor)
```

Comment:
It would be prefferable if this logic didn't have to be duplicated in two places